File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/product/product_mst_dto.cpp
#include "product_mst_dto.h"
#include <iostream>
using namespace std;

ProductMstDTO::ProductMstDTO(long product_mst_id, string product_type, string product_name, double price, PriceUnitType price_unit) {
	this->product_mst_id = product_mst_id;
	this->product_type = product_type;
	this->product_name = product_name;
	this->price = price;
	this->price_unit = price_unit;
}

long &ProductMstDTO::get_product_mst_id() {
	return product_mst_id;
}

string &ProductMstDTO::get_product_type() {
	return product_type;
}

string &ProductMstDTO::get_product_name() {
	return product_name;
}

double &ProductMstDTO::get_price() {
    return price;
}

PriceUnitType &ProductMstDTO::get_price_unit() {
    return price_unit;
}

ostream &operator <<(ostream &out, ProductMstDTO &obj) {
	out << obj.product_mst_id << " " << obj.product_type << " "
			<< obj.product_name << " " << obj.price << " " << obj.price_unit;
	return out;
}

istream &operator >>(istream &in, ProductMstDTO &obj) {
    string price_unit;
	in >> obj.product_mst_id >> obj.product_type >> obj.product_name >> obj.price >> price_unit;
	obj.price_unit = price_unit_type_value_of_string(price_unit);
	return in;
}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/product/ProductMstDto.java
package jp.co.worksap.intern.entities.product;

import jp.co.worksap.intern.constants.PriceUnitType;
import jp.co.worksap.intern.entities.ICsvMasterDTO;

public class ProductMstDto implements ICsvMasterDTO {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1950118052473203296L;
	private Long productMstId;
	private String productType;
	private String productName;
	private double price;
	private PriceUnitType priceUnit;

	/**
	 * Default Constructor
	 * 
	 * @param productMstId
	 * @param productType
	 * @param productName
	 * @param price
	 * @param priceUnit
	 */
	public ProductMstDto(Long productMstId, String productType, String productName, double price, PriceUnitType priceUnit) {
		super();
		this.productMstId = productMstId;
		this.productType = productType;
		this.productName = productName;
		this.price = price;
		this.priceUnit = priceUnit;
	}

	public Long getProductMstId() {
		return productMstId;
	}

	public String getProductType() {
		return productType;
	}

	public String getProductName() {
		return productName;
	}

	public double getPrice() {
		return price;
	}

	public PriceUnitType getPriceUnit() {
		return priceUnit;
	}

	

}